src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable no-lone-blocks */
+/* eslint-disable no-console */
+/* eslint-disable quote-props */
+import cn from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  deleteTodo: (arg: number) => void;
+  updateTodo: (arg: number, obj: Partial<Todo>) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo, deleteTodo, updateTodo,
+}) => {
+  const { id, title, completed } = todo;
+  const [update, setUpdate] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const editHandler = (e: React.FocusEvent<HTMLInputElement, Element>) => {
+    if (e.target.value.trim()) {
+      updateTodo(id, { title: e.target.value.trim() });
+      setUpdate(false);
+    } else {
+      deleteTodo(id);
+    }
+  };
+
+  useEffect(() => {
+    if (update && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [update]);
+
+  return (
+    <div className={cn('todo', { completed })}>
+      <label
+        className={cn('todo__status-label', { outline: !completed })}
+      >
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onChange={() => updateTodo(id, { completed: !completed })}
+        />
+      </label>
+      {!update
+        ? (
+          <>
+            <span
+              className="todo__title"
+              onDoubleClick={() => setUpdate(true)}
+            >
+              {title}
+            </span>
+            <button
+              type="button"
+              className="todo__remove"
+              onClick={() => deleteTodo(id)}
+            >
+              ×
+            </button>
+          </>
+        ) : (
+          <>
+            <form>
+              <input
+                type="text"
+                className="todo__title-field"
+                placeholder="Empty todo will be deleted"
+                defaultValue={title}
+                onBlur={editHandler}
+                ref={inputRef}
+              />
+            </form>
+
+            <div className="modal overlay">
+              <div className="modal-background has-background-white-ter" />
+              <div className="loader" />
+            </div>
+          </>
+        ) }
+    </div>
+  );
+};
+
+{
+  <div className="todo completed">
+    <label className="todo__status-label">
+      <input type="checkbox" className="todo__status" checked />
+    </label>
+
+    <span className="todo__title">Completed Todo</span>
+
+    {/* Remove button appears only on hover */}
```

Comment:
Remove comments. Fix all cases. Try it write obvious code that explain all we need without comments)